src/NexusMods.Networking.GitHub/GitHubApi.cs

```diff
@@ -0,0 +1,105 @@
+using System.Diagnostics.CodeAnalysis;
+using System.Net;
+using System.Net.Http.Headers;
+using System.Net.Http.Json;
+using JetBrains.Annotations;
+using Microsoft.Extensions.Logging;
+using NexusMods.Networking.GitHub.DTOs;
+
+namespace NexusMods.Networking.GitHub;
+
+[PublicAPI]
+internal class GitHubApi : IGitHubApi
+{
+    private const string BaseUrl = "https://api.github.com";
+
+    // https://docs.github.com/en/rest/about-the-rest-api/api-versions
+    private const string HeaderApiVersion = "X-GitHub-Api-Version";
+    private const string ApiVersion = "2022-11-28";
+
+    private readonly ILogger _logger;
+    private readonly HttpClient _httpClient;
+
+    public AuthenticationHeaderValue? AuthenticationHeaderValue { get; set; }
```

Comment:
Can you add some comments for this? It is only used (or rather currently unused) in the tests.
Clarify for what it is supposed to be used and when it is used.